pkg/local_object_storage/metabase/list_test.go

```diff
@@ -0,0 +1,173 @@
+		err = putBig(db, obj.Object())
+		require.NoError(t, err)
+		expected = append(expected, obj.Object().Address())
+
+		// add one inhumed (do not include into expected)
+		obj = generateRawObjectWithCID(t, containerID)
+		obj.SetType(objectSDK.TypeRegular)
+		err = putBig(db, obj.Object())
+		require.NoError(t, err)
+		ts := generateRawObjectWithCID(t, containerID)
+		err = meta.Inhume(db, obj.Object().Address(), ts.Object().Address())
+		require.NoError(t, err)
+
+		// add one child object (do not include parent into expected)
+		splitID := objectSDK.NewSplitID()
+		parent := generateRawObjectWithCID(t, containerID)
+		addAttribute(parent, "foo", "bar")
+		child := generateRawObjectWithCID(t, containerID)
+		child.SetParent(parent.Object().SDK())
+		child.SetParentID(parent.ID())
+		child.SetSplitID(splitID)
+		err = putBig(db, child.Object())
+		require.NoError(t, err)
+		expected = append(expected, child.Object().Address())
+	}
+
+	expected = sortAddresses(expected)
+
+	t.Run("success with various count", func(t *testing.T) {
+		for countPerReq := 1; countPerReq <= total; countPerReq++ {
+			got := make([]*objectSDK.Address, 0, total)
+
+			res, cursor, err := meta.ListWithCursor(db, uint32(countPerReq), "")
+			require.NoError(t, err, "count:%d", countPerReq)
+			got = append(got, res...)
+
+			expectedIterations := total / countPerReq
+			if total%countPerReq == 0 { // remove initial list if aligned
+				expectedIterations--
+			}
+
+			for i := 0; i < expectedIterations; i++ {
+				res, cursor, err = meta.ListWithCursor(db, uint32(countPerReq), cursor)
+				require.NoError(t, err, "count:%d", countPerReq)
+				got = append(got, res...)
+			}
+
+			_, _, err = meta.ListWithCursor(db, uint32(countPerReq), cursor)
+			require.ErrorIs(t, err, meta.ErrEndOfListing, "count:%d", countPerReq, cursor)
+
+			got = sortAddresses(got)
+			require.Equal(t, expected, got, "count:%d", countPerReq)
+		}
+	})
+
+	t.Run("invalid cursor", func(t *testing.T) {
+		_, cursor, err := meta.ListWithCursor(db, total/2, "")
+		require.NoError(t, err)
+
+		_, _, err = meta.ListWithCursor(db, total/2, "x"+cursor[1:])
+		require.Error(t, err)
+
+		_, _, err = meta.ListWithCursor(db, total/2, cursor[1:])
+		require.Error(t, err)
+	})
+
+	t.Run("invalid count", func(t *testing.T) {
+		_, _, err := meta.ListWithCursor(db, 0, "")
+		require.ErrorIs(t, err, meta.ErrEndOfListing)
+	})
+}
+
+func TestAddObjectDuringListingWithCursor(t *testing.T) {
+	db := newDB(t)
+
+	const total = 5
+
+	expected := make(map[string]int, total)
+
+	// fill metabase with objects
+	for i := 0; i < total; i++ {
+		obj := generateRawObject(t)
+		err := putBig(db, obj.Object())
+		require.NoError(t, err)
+		expected[obj.Object().Address().String()] = 0
+	}
+
+	// get half of the objects
+	got, cursor, err := meta.ListWithCursor(db, total/2, "")
+	require.NoError(t, err)
+	for _, obj := range got {
+		if _, ok := expected[obj.String()]; ok {
+			expected[obj.String()] = 1
+		}
+	}
```

Comment:
We also want to test that objects don't appear twice. This is a real possibility, for example when newly added objects were added behind the current cursor. So we could raise an error if `expected[obj.String()]` is already 1, or do `++` and check `== 1` in the end.